pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -212,32 +212,37 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus) *kubecontainer.Runtim
 	return &kubecontainer.RuntimeStatus{Conditions: conditions}
 }
 
-func fieldProfile(scmp *v1.SeccompProfile, profileRootPath string, fallbackToRuntimeDefault bool) string {
+func fieldProfile(scmp *v1.SeccompProfile, profileRootPath string, fallbackToRuntimeDefault bool) (string, error) {
 	if scmp == nil {
 		if fallbackToRuntimeDefault {
-			return v1.SeccompProfileRuntimeDefault
+			return v1.SeccompProfileRuntimeDefault, nil
 		}
-		return ""
+		return "", nil
 	}
 	if scmp.Type == v1.SeccompProfileTypeRuntimeDefault {
-		return v1.SeccompProfileRuntimeDefault
+		return v1.SeccompProfileRuntimeDefault, nil
 	}
-	if scmp.Type == v1.SeccompProfileTypeLocalhost && scmp.LocalhostProfile != nil && len(*scmp.LocalhostProfile) > 0 {
-		fname := filepath.Join(profileRootPath, *scmp.LocalhostProfile)
-		return v1.SeccompLocalhostProfileNamePrefix + fname
+	if scmp.Type == v1.SeccompProfileTypeLocalhost {
+		if scmp.LocalhostProfile != nil && len(*scmp.LocalhostProfile) > 0 {
+			fname := filepath.Join(profileRootPath, *scmp.LocalhostProfile)
+			return v1.SeccompLocalhostProfileNamePrefix + fname, nil
+		} else {
+			err := fmt.Errorf("localhostProfile must be set if seccompProfile type is Localhost.")
+			return "", err
+		}
 	}
 	if scmp.Type == v1.SeccompProfileTypeUnconfined {
-		return v1.SeccompProfileNameUnconfined
+		return v1.SeccompProfileNameUnconfined, nil
 	}
 
 	if fallbackToRuntimeDefault {
```

Comment:
should the unsupported `scmp.Type` also lead to the error?

Reply:
Do you mean for example, if the `scmp.Type` is something undefined?

Reply:
no, the value other than those three above: `SeccompProfileTypeUnconfined `, etc.

Reply:
I'm sorry, I think I may be misunderstanding your question. If localhostProfile is set at all for those other profile types, it will already return an error during validation.

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L4270

Reply:
yes. I also wasn't clear. It is just a fool proof aspect of the code at this point. If we will add a new type at some point I want code to fail unless updated. Also it will not run for static pods and I'm not sure if we have kubelet admission logic checking for the same constraint

Reply:
I think I understand now - are you suggesting like a conditional that will just catch any other scmp.Type and return a (different) error about an unsupported type?

Reply:
tried to think of a graceful way to handle this in both affected functions. can you please take another look?